File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.cpp
class Solution {
public:
    bool isOperator(string s){
        return (s == "+" || s == "-" || s == "*" || s == "/");
    }
    string helper(string a, string b, string op){
        int x = stoi(a);
        int y = stoi(b);
        
        int res;
        
        if(op == "+")   res = x + y;
        if(op == "-")   res = x - y;
        if(op == "*")   res = x * y;
        if(op == "/")   res = x / y;
        
        return to_string(res);
    }
    int evalRPN(vector<string>& tokens) {
        stack<string>st;
        
        for(int i = 0; i < tokens.size(); i++){
            if(!isOperator(tokens[i])){
                st.push(tokens[i]);
            }else{
                string s1 = st.top();
                st.pop();
                string s2 = st.top();
                st.pop();
                
                string res = helper(s2, s1, tokens[i]);
                st.push(res);
                
            }
        }
        
        return stoi(st.top());
    }
};

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
class Solution {
    public boolean isOperator(String s) {
        return (s.equals("+") || s.equals("-") || s.equals("*") || s.equals("/"));
    }

    public String helper(String a, String b, String op) {
        int x = Integer.parseInt(a);
        int y = Integer.parseInt(b);

        int res = 0;

        if (op.equals("+")) res = x + y;
        if (op.equals("-")) res = x - y;
        if (op.equals("*")) res = x * y;
        if (op.equals("/")) res = x / y;

        return Integer.toString(res);
    }

    public int evalRPN(String[] tokens) {
        Stack<String> st = new Stack<>();

        for (int i = 0; i < tokens.length; i++) {
            if (!isOperator(tokens[i])) {
                st.push(tokens[i]);
            } else {
                String s1 = st.pop();
                String s2 = st.pop();

                String res = helper(s2, s1, tokens[i]);
                st.push(res);
            }
        }

        return Integer.parseInt(st.pop());
    }
}